internal/datanode/compactor/backfill_compactor.go

```diff
@@ -0,0 +1,495 @@
+	if err != nil {
+		return nil, "", err
+	}
+	statsPath := metautil.JoinIDPath(collectionID, partitionID, segmentID, outputFieldID, statsID)
+	bm25LogPath := path.Join(cli.RootPath(), common.SegmentBm25LogPath, statsPath)
+
+	bytes, err := stats.Serialize()
+	if err != nil {
+		return nil, "", err
+	}
+	err = cli.Write(t.ctx, bm25LogPath, bytes)
+	if err != nil {
+		return nil, "", err
+	}
+
+	// Build bm25LogPathForResult
+	bm25LogPathForResult := metautil.BuildBm25LogPath(
+		t.compactionParams.StorageConfig.GetRootPath(),
+		collectionID, partitionID, segmentID, outputFieldID, statsID)
+	// For non-local storage, prepend bucket name to match the truePaths pattern
+	if t.compactionParams.StorageConfig.GetStorageType() != "local" {
+		bm25LogPathForResult = path.Join(t.compactionParams.StorageConfig.GetBucketName(), bm25LogPathForResult)
+	}
+
+	return bytes, bm25LogPathForResult, nil
+}
+
+func (t *backfillCompactionTask) buildMergedLogs(segment *datapb.CompactionSegmentBinlogs, newColumnGroups []storagecommon.ColumnGroup, paths, truePaths []string, sparseFieldMemorySize int, inputStrs []string, bytes []byte, bm25LogPathForResult string, outputFieldID int64) ([]*datapb.FieldBinlog, []*datapb.FieldBinlog, error) {
+	// build new InsertLogs from writer
+	newInsertLogs := make(map[int64]*datapb.FieldBinlog)
+	for i, columnGroup := range newColumnGroups {
+		truePath := truePaths[i]
+		fileSize, err := packed.GetFileSize(truePath, t.compactionParams.StorageConfig)
+		if err != nil {
+			return nil, nil, err
+		}
+		// Use the path without bucket prefix for LogPath
+		logPath := paths[i]
+		if t.compactionParams.StorageConfig.GetStorageType() != "local" {
+			logPath = truePath
+		}
+		fieldBinlog := &datapb.FieldBinlog{
+			FieldID:     columnGroup.GroupID,
+			ChildFields: columnGroup.Fields,
+			Binlogs: []*datapb.Binlog{
+				{
+					LogSize:    fileSize,
+					MemorySize: int64(sparseFieldMemorySize), // For packed format, compressed size equals memory size
+					LogPath:    logPath,
+					EntriesNum: int64(len(inputStrs)),
+				},
+			},
+		}
+		newInsertLogs[columnGroup.GroupID] = fieldBinlog
+	}
+
+	// build new Bm25Logs
+	bm25FileSize := int64(len(bytes))
+	newBm25Logs := []*datapb.FieldBinlog{
+		{
+			FieldID: outputFieldID,
+			Binlogs: []*datapb.Binlog{
+				{
+					LogSize:    bm25FileSize,
+					MemorySize: bm25FileSize,
+					LogPath:    bm25LogPathForResult,
+					EntriesNum: int64(len(inputStrs)),
+				},
+			},
+		},
+	}
+
+	// merge with original segment's binlogs
+	originalInsertLogs := segment.GetFieldBinlogs()
+	mergedInsertLogs := make([]*datapb.FieldBinlog, 0, len(originalInsertLogs)+len(newInsertLogs))
+	mergedInsertLogs = append(mergedInsertLogs, originalInsertLogs...)
+	newInsertLogsList := storage.SortFieldBinlogs(newInsertLogs)
+	mergedInsertLogs = append(mergedInsertLogs, newInsertLogsList...)
+	mergedBm25Logs := newBm25Logs
+
+	return mergedInsertLogs, mergedBm25Logs, nil
+}
+
```

Comment:
main process for executing bm25 backfill process
read only one column and write only one column
without violating data in other fields